### PR TITLE
Display hostName on the rollbar

### PR DESCRIPTION
**Pull request by apatil1401**

### CODE REVIEW

**Review by akumar39 — CHANGES_REQUESTED** [review]
Add .gitignore to ignore all intellij generated files - 
.idea
*.iml

File: src/main/java/com/zendesk/rollbar/RollbarLog4jAppender.java [hunk 1]
```diff
@@ -27,34 +27,42 @@
...
+        this.client = Rollbar.init(config);
       }
-      else {
-        config = ConfigBuilder.withAccessToken(this.accessToken)
-                .environment(this.environment)
-                .build();
-      }
-      this.client = Rollbar.init(config);
+    }catch(Exception e){
+      LogLog.error("Invalid hostName");
```
Comment (akumar39) [inline src/main/java/com/zendesk/rollbar/RollbarLog4jAppender.java]: the try ;catch blocks don't add up to catch the right exception. Make sure your try{} is relevant to the catch block. It looks like you're logging for invalid hostname but your try does more than that

File: src/main/java/com/zendesk/rollbar/RollbarLog4jAppender.java [hunk 2]
```diff
@@ -27,27 +27,35 @@
...
   public synchronized void activateOptions() {
     super.activateOptions();
+    try {
+      InetAddress ip = InetAddress.getByName(hostName);
+    }catch(Exception e){
+      LogLog.error("Invalid hostName");
+    }
     if (this.accessToken != null && !this.accessToken.isEmpty() && this.environment != null
-        && !this.environment.isEmpty()) {
+            && !this.environment.isEmpty() && !this.hostName.isEmpty() && this.hostName != null) {
```
Comment (akumar39) [inline src/main/java/com/zendesk/rollbar/RollbarLog4jAppender.java]: hostname is not being used in this block - so remove the check

**Review by akumar39 — CHANGES_REQUESTED** [review]
For consistency, add the same changes to the other appender as well

File: src/main/java/com/zendesk/rollbar/RollbarLog4jAppender.java (hunk 2 above)
Comment (apatil1401) [reply src/main/java/com/zendesk/rollbar/RollbarLog4jAppender.java]: Okay

File: src/main/java/com/zendesk/rollbar/RollbarLog4j2Appender.java [hunk 3]
```diff
@@ -77,6 +87,7 @@ public static RollbarLog4j2Appender createAppender(@PluginAttribute("name") Stri
       layout = PatternLayout.createDefaultLayout();
     }
 
+
```
Comment (akumar39) [inline src/main/java/com/zendesk/rollbar/RollbarLog4j2Appender.java]: Remove this blank line

File: src/main/java/com/zendesk/rollbar/RollbarLog4jAppender.java [hunk 4]
```diff
@@ -27,27 +27,35 @@
 
 import static com.rollbar.api.payload.data.Level.CRITICAL;
 import static org.apache.log4j.Level.*;
-
+i
```
Comment (akumar39) [inline src/main/java/com/zendesk/rollbar/RollbarLog4jAppender.java]: Typo - please remove

File: src/main/java/com/zendesk/rollbar/RollbarLog4jAppender.java [hunk 5]
```diff
@@ -27,27 +27,35 @@
...
 
 public class RollbarLog4jAppender extends AppenderSkeleton {
 
-  private String accessToken;
+  private String accessToken;mport java.util.*;
+import java.net.*;
```
Comment (akumar39) [inline src/main/java/com/zendesk/rollbar/RollbarLog4jAppender.java]: Formatting

Comment (akumar39) [reply src/main/java/com/zendesk/rollbar/RollbarLog4jAppender.java]: Indent along the lines of the rest of the code

File: src/main/java/com/zendesk/rollbar/RollbarLog4jAppender.java [hunk 6]
```diff
@@ -27,27 +27,35 @@
...
   public synchronized void activateOptions() {
     super.activateOptions();
+    try {
+      InetAddress ip = InetAddress.getByName(hostName);
+    }catch(Exception e){
+      LogLog.error("Invalid hostName");
+    }
     if (this.accessToken != null && !this.accessToken.isEmpty() && this.environment != null
-        && !this.environment.isEmpty()) {
+            && !this.environment.isEmpty() ) {
```
Comment (akumar39) [inline src/main/java/com/zendesk/rollbar/RollbarLog4jAppender.java]: remove the extra spaces

File: src/main/java/com/zendesk/rollbar/RollbarLog4jAppender.java [hunk 7]
```diff
@@ -90,17 +107,19 @@ protected void append(LoggingEvent event) {
       return;
     }
 
+
```
Comment (akumar39) [inline src/main/java/com/zendesk/rollbar/RollbarLog4jAppender.java]: Remove the blank line

**Review by akumar39 — COMMENTED** [review]
Mostly good, just some indentation and formatting changes

**Review by akumar39 — APPROVED** [review]
Looks good

**Pull request merged**